### PR TITLE
chore: fix some typos in comments

### PR DESCRIPTION
**Pull request by goodactive**

### CODE REVIEW

**Review by wader — APPROVED** [review]
We already fixed some stuff in vendored code (decnumber) in #3060 so i guess it's fine

**Comment by goodactive** [comment]
> We already fixed some stuff in vendored code (decnumber) in #3060 so i guess it's fine

Thanks for the replay.

**Comment by itchyny** [comment]
But, did you try sending the patch to the original author? It is recommended to send corrections to the upstream in general, otherwise these changes may be reverted when we follow the upstream for any security reasons. The E-mail address is written in the first comment in each file of decNumber.

**Comment by emanuele6** [comment]
Yes, none of those typos are in jq code; the one in `src/util.c` is from [res2001/strptime](https://github.com/res2001/strptime), and the other ones are all from [decNumber](https://speleotrove.com/decimal/decnumber.html).
It would make a lot more sense to report those typos upstreams, since we generally do not care about reading the content of those files, and we don't want to modify them unless we have to...

File: src/decNumber/decNumber.c [hunk 1]
```diff
@@ -3798,7 +3798,7 @@ static void decToString(const decNumber *dn, char *string, Flag eng) {
 /*                                                                    */
 /* Addition, especially x=x+1, is speed-critical.                     */
 /* The static buffer is larger than might be expected to allow for    */
-/* calls from higher-level funtions (notable exp).                    */
```
Comment (emanuele6) [inline src/decNumber/decNumber.c:3801]: You broke the comment box fixing the typo without removing a space

File: src/decNumber/decNumber.c [hunk 2]
```diff
@@ -4212,7 +4212,7 @@ static decNumber * decAddOp(decNumber *res, const decNumber *lhs,
 /* long subtractions.  These are acc and var1 respectively.           */
 /* var1 is a copy of the lhs coefficient, var2 is the rhs coefficient.*/
 /* The static buffers may be larger than might be expected to allow   */
-/* for calls from higher-level funtions (notable exp).                */
```
Comment (emanuele6) [inline src/decNumber/decNumber.c:4215]: same here

File: src/decNumber/decNumber.c [hunk 3]
```diff
@@ -5203,7 +5203,7 @@ static decNumber * decMultiplyOp(decNumber *res, const decNumber *lhs,
 /*    exp(-x) where x can be the tiniest number (Ntiny).              */
 /*                                                                    */
 /* 2. Normalizing x to be <=0.1 (instead of <=1) reduces loop         */
-/*    iterations by appoximately a third with additional (although    */
```
Comment (emanuele6) [inline src/decNumber/decNumber.c:5206]: same here

**Pull request merged**